### PR TITLE
fix(client-s3-control): restrict Bucket param dedupe to S3 only

### PR DESCRIPTION
**Pull request by kuhe**
*Issue #, if available:*
internal V750691458

*Description of changes:*
restricts the context param Bucket and uri segment Bucket deduplication to only S3.

### CODE REVIEW

File: codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsRestXml.java [hunk 1]
```diff
@@ -368,6 +371,20 @@ private void deserializeDocumentBody(
         }
     }
 
+    @Override
+    public void generateRequestSerializers(GenerationContext context) {
+        String serviceId = context.getService()
+            .getTrait(ServiceTrait.class)
+            .map(ServiceTrait::getSdkId)
+            .orElse("");
+        if (serviceId.equals("S3")) {
```
Comment (kuhe) [inline codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsRestXml.java:380]:
```suggestion

if (serviceId.equals("S3")) {
```

File: codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsRestXml.java [hunk 2]
```diff
@@ -368,6 +371,21 @@ private void deserializeDocumentBody(
...
     }
 
+    @Override
+    public void generateRequestSerializers(GenerationContext context) {
+        String serviceId = context.getService()
+            .getTrait(ServiceTrait.class)
+            .map(ServiceTrait::getSdkId)
+            .orElse("");
+
+if (serviceId.equals("S3")) {
```
Comment (kuhe) [inline codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsRestXml.java:381]:
```suggestion
        if (serviceId.equals("S3")) {
```

File: codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsRestXml.java [hunk 3]
```diff
@@ -17,8 +17,11 @@
 
 import static software.amazon.smithy.aws.typescript.codegen.propertyaccess.PropertyAccessor.getFrom;
 
+import java.util.Collections;
 import java.util.List;
 import java.util.Set;
+
```
Comment (kuhe) [inline codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsRestXml.java:23]:
```suggestion
```

**Comment by kuhe** [comment]
https://github.com/awslabs/smithy-typescript/pull/626

**Review by trivikr — APPROVED** [review]

**Pull request merged**